### PR TITLE
Refactor helper classes to be much cleaner.

### PR DESCRIPTION
**Pull request by samtstern**
Also fix the bug where framework Activities did not work (#124)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,10 +1,11 @@
 buildscript {
     repositories {
         jcenter()
+        maven { url 'https://maven.google.com' }
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.4.0-alpha7'
+        classpath 'com.android.tools.build:gradle:3.0.0-alpha1'
```
Comment (SUPERCILEX) [inline build.gradle]: alpha 3 is available now 😄

File: build.gradle [hunk 2]
```diff
@@ -14,6 +15,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        maven { url 'https://maven.google.com' }
```
Comment (SUPERCILEX) [inline build.gradle]: If you use `gradle-4.0-rc-3-all.zip` in the Gradle wrapper, `google()` is available which looks way cleaner. 😄

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Thu May 11 19:29:35 PDT 2017
+#Tue Jun 13 11:26:34 PDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-milestone-1-all.zip
```
Comment (SUPERCILEX) [inline gradle/wrapper/gradle-wrapper.properties]: Yeah, run `./gradlew wrapper --gradle-version=4.0-rc-3`

**Comment by SUPERCILEX** [comment]
@samtstern 👌 💯

**Comment by samtstern** [comment]
@SUPERCILEX our tests fail with anything higher than `3.0.0-alpha1` and in spectacularly different ways!  I will leave figuring that out for another day.

**Comment by SUPERCILEX** [comment]
@samtstern Aw crud. 😢

**Pull request merged**